api/access/server.go

```diff
@@ -0,0 +1,515 @@
+	blockID := flow.HashToID(in.Id)
+	height, err := s.index.HeightForBlock(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get height for block %x: %w", blockID, err)
+	}
+
+	req := access.GetBlockHeaderByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockHeaderByHeight(ctx, &req)
+}
+
+func (s *Server) GetBlockHeaderByHeight(_ context.Context, in *access.GetBlockHeaderByHeightRequest) (*access.BlockHeaderResponse, error) {
+	header, err := s.index.Header(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block header at height %d: %w", in.Height, err)
+	}
+
+	blockMsg, err := convert.BlockHeaderToMessage(header)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert block header to RPC entity: %w", err)
+	}
+
+	resp := access.BlockHeaderResponse{
+		Block: blockMsg,
+	}
+
+	return &resp, err
+}
+
+func (s *Server) GetLatestBlock(ctx context.Context, in *access.GetLatestBlockRequest) (*access.BlockResponse, error) {
+	height, err := s.index.Last()
+	if err != nil {
+		return nil, fmt.Errorf("could not get last height: %w", err)
+	}
+
+	req := &access.GetBlockByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockByHeight(ctx, req)
+}
+
+func (s *Server) GetBlockByID(ctx context.Context, in *access.GetBlockByIDRequest) (*access.BlockResponse, error) {
+	blockID := flow.HashToID(in.Id)
+	height, err := s.index.HeightForBlock(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get height for block %x: %w", blockID, err)
+	}
+
+	req := access.GetBlockByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockByHeight(ctx, &req)
+}
+
+func (s *Server) GetBlockByHeight(_ context.Context, in *access.GetBlockByHeightRequest) (*access.BlockResponse, error) {
+	header, err := s.index.Header(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get header for height %d: %w", in.Height, err)
+	}
+
+	sealIDs, err := s.index.SealsByHeight(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get seals for height %d: %w", in.Height, err)
+	}
+
+	seals := make([]*entities.BlockSeal, 0, len(sealIDs))
+	for _, sealID := range sealIDs {
+		seal, err := s.index.Seal(sealID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get seal with ID %x: %w", sealID, err)
+		}
+
+		blockID := seal.BlockID
+		resultID := seal.ResultID
+
+		// See https://github.com/onflow/flow-go/blob/v0.17.4/engine/common/rpc/convert/convert.go#L180-L188
+		entity := entities.BlockSeal{
+			BlockId:                    blockID[:],
+			ExecutionReceiptId:         resultID[:],
+			ExecutionReceiptSignatures: [][]byte{}, // filling seals signature with zero
+		}
+		seals = append(seals, &entity)
+	}
+
+	collIDs, err := s.index.CollectionsByHeight(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get collections for height %d: %w", in.Height, err)
+	}
+
+	collections := make([]*entities.CollectionGuarantee, 0, len(collIDs))
```

Comment:
Same, could also be `collectionMsgs`. In fact, if we want to be consistent, only `flow.XXX` objects should be called without additional suffix, and `entities.XXX` should then all be `XXXMsg`.